### PR TITLE
[Diagnostics] Finish porting subscript errors

### PR DESCRIPTION
**Pull request by hborla**
* Diagnose subscripting a nil literal in CSGen
* Diagnose ambiguities due to missing arguments in `diagnoseAmbiguityWithFixes`
* Remove `diagnoseSubscriptErrors` from CSDiag

### CODE REVIEW

**Comment by hborla** [comment]
@swift-ci please smoke test

**Review by xedin — APPROVED** [review]
This is great, thank you!

**Comment by hborla** [comment]
@swift-ci please smoke test

**Comment by hborla** [comment]
@swift-ci please smoke test

**Comment by hborla** [comment]
@swift-ci please smoke test

File: lib/Sema/CSGen.cpp [hunk 1]
```diff
@@ -930,6 +930,16 @@ namespace {
       = { nullptr, nullptr };
     unsigned currentEditorPlaceholderVariable = 0;
 
+    /// Returns false and emits the specified diagnostic if the member reference
+    /// base is a nil literal. Returns true otherwise.
+    bool isValidBaseOfMemberRef(Expr *base, Diag<> diagnostic) {
+      if (auto nilLiteral = dyn_cast<NilLiteralExpr>(base)) {
```
Comment (slavapestov) [inline lib/Sema/CSGen.cpp]: It's odd that we have this special case for subscripts but not other member references. Maybe we should just remove this and possibly handle the diagnostic elsewhere?

Comment (xedin) [reply lib/Sema/CSGen.cpp]: I mentioned that in my previous comment - we are going to extend that to all references soon.

Comment (slavapestov) [reply lib/Sema/CSGen.cpp]: Ah, I didn't see that, my bad.

Comment (xedin) [reply lib/Sema/CSGen.cpp]: No problem :)

**Pull request merged**